terraform/shared/video-player/js/common.js

```diff
@@ -57,20 +58,45 @@ function httpStream() {
   loadStream();
 }
 
+//Video stage counter
+var streamState = 0;
+
 function ipfsStream() {
   live.src({
     src: stream_url_ipfs,
     type: stream_type
   });
   loadStream();
   videojs.Hls.xhr.beforeRequest = function(options) {
-    // Replace IPFS gateway of origin with that of this node
-    if (options.uri.indexOf('/ipfs/')>0) {
+
+    //First hit is m3u8, start playing
+    if (options.uri.indexOf('.m3u8') > 0) {
+      if (!streamState) {
+        live.play();
+        streamState = 1;
+      }
+    }
+
+    if (options.uri.indexOf('/ipfs/') > 0) {
       document.getElementById('loadingTitle').innerHTML = 'Located stream via IPFS';
       document.getElementById('msg').innerHTML = 'Downloading video content...';
-      options.uri = ipfs_gateway + options.uri.substring(options.uri.indexOf('/ipfs/'));
+      // Replace IPFS gateway of origin with that of this node
+      options.uri = ipfs_gateway_origin + options.uri.substring(options.uri.indexOf('/ipfs/'));

```

Comment:
Why is this origin and not specified gateway? It also does not match the comment above